pages/data-providers/[data-provider-id]/plugins/index.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react'
+
+import pluginsClassNames from './plugins.css'
+
+import { withGlobalStore } from 'store'
+import { Card, Button } from 'design'
+import { overview as texts } from 'texts/plugins'
+
+const PluginCard = ({
+  title,
+  description,
+  enabled,
+  onEnable,
+  href,
+  ...restProps
+}) => {
+  const [isLoading, setLoading] = useState(false)
+  const handleEnable = event => {
+    if (onEnable) {
+      onEnable(event)
+      setLoading(true)
+    }
+  }
+
+  // TODO: If there is a loading state we have to automatically perform
+  // route change after enabling
```

Comment:
@Joozty I am struggling to do this, and struggling much more to do this minimising side effects.

The experience should be the following:
1. A user visits `/plugins` page
2. There is a _Discovery_ plugin that is not enabled yet
3. The user presses _Enable_ and sees button changed state to loading
4. After successful enabling (generating a key) the user should see the instructions on the another page

Ideally the router call should be implicit but if it's not possible, explicit is fine too.

Reply:
This seems to be wrong. I will rethink the process.

Reply:
You are looking probably for something like this

import React, { useState, useEffect, useCallback } from 'react'

```
const PluginCard = ({
  title,
  description,
  enabled,
  onEnable,
  href,
  ...restProps
}) => {
  const [isLoading, setLoading] = useState(false)

  // memoize this function, i.e. don't recalculate it on every render
  const handleEnable = useCallback(async event => {
    if (onEnable) {
      try {
        await onEnable(event)
        setLoading(true)
      } catch (ignoreError) {
        // ignoring API error
      }

    }
  }, [])

  useEffect(() => {
    // false -> true
    if (isLoading) {
      // handle route change here
    }
  }, [isLoading]) // this function is called only when isLoading value changes

  return (
    ...
  )
}

```

Reply:
@mcancellieri this user experience is caused by back-end and it's not really good. I see the following options:
1. Generate the key implicitly when the user visits instructions page
2. Generate keys implicitly when the user visits generic plugins page
3. Generate keys implicitly once user registers in the Dashboard

Reply:
yes that's fine. In the previous dashboard we generated the key as soon as the user opened the instructions tab... would that work?

Reply:
I you can generate the key first time `/plugins` endpoint is requested responding always with keys without affects on performance, it would be amazing.

@valerapro FYI

Reply:
Yes, it possible generate key for first time request.